File: zavrsni-rad-2/src/main/java/hr/vsite/igavlik/zavrsnirad/dto/v3/PoissonCard.java
package hr.vsite.igavlik.zavrsnirad.dto.v3;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PoissonCard {
    private String homeTeamName;
    private int homeTeamGoalGuess;
    private double homeTeamLambda;
    private double homeTeamHomeAttackFactor;
    private double  homeTeamHomeDefenseFactor;

    private String guestTeamName;
    private int  guestTeamGoalGuess;
    private double guestTeamLambda;
    private double guestTeamGuestAttackFactor;
    private double guestTeamGuestDefenseFactor;

    private double probabilityOfGuess;

    private double homeTeamProbabilityOf0;
    private double homeTeamProbabilityOf1;
    private double homeTeamProbabilityOf2;
    private double homeTeamProbabilityOf3;
    private double homeTeamProbabilityOf4;
    private double homeTeamProbabilityOf5;

    private double guestTeamProbabilityOf0;
    private double guestTeamProbabilityOf1;
    private double guestTeamProbabilityOf2;
    private double guestTeamProbabilityOf3;
    private double guestTeamProbabilityOf4;
    private double guestTeamProbabilityOf5;

    public PoissonCard(TeamReview homeTeam, TeamReview guestTeam) {
        this.homeTeamName = homeTeam.getName();
        this.homeTeamHomeAttackFactor = homeTeam.getAsHomeAttackFactor();
        this.homeTeamHomeDefenseFactor = homeTeam.getAsHomeDefensekFactor();

        this.guestTeamName = guestTeam.getName();
        this.guestTeamGuestAttackFactor = guestTeam.getAsGuestAttackFactor();
        this.guestTeamGuestDefenseFactor = guestTeam.getAsGuestDefensekFactor();
    }
}

File: 2020_05_04_Zavrsni_Frontend_2/zavrsni-frontend/src/app/components/poisson-card/poisson-card.ts
export interface PoissonCard {
  homeTeamName: string;
  homeTeamGoalGuess: number;
  homeTeamLambda: number;
  homeTeamHomeAttackFactor: number;
  homeTeamHomeDefenseFactor: number;

  guestTeamName: string;
  guestTeamGoalGuess: number;
  guestTeamLambda: number;
  guestTeamGuestAttackFactor: number;
  guestTeamGuestDefenseFactor: number;

  probabilityOfGuess: number;

  // details panel
  homeTeamProbabilityOf0: number;
  homeTeamProbabilityOf1: number;
  homeTeamProbabilityOf2: number;
  homeTeamProbabilityOf3: number;
  homeTeamProbabilityOf4: number;
  homeTeamProbabilityOf5: number;

  guestTeamProbabilityOf0: number;
  guestTeamProbabilityOf1: number;
  guestTeamProbabilityOf2: number;
  guestTeamProbabilityOf3: number;
  guestTeamProbabilityOf4: number;
  guestTeamProbabilityOf5: number;

}
